benchmark/BenchmarkToJson.h

```diff
@@ -0,0 +1,43 @@
+// Copyright 2023, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Andre Schlegel February of 2023, schlegea@informatik.uni-freiburg.de)
+
+#pragma once
+
+#include <vector>
+#include <utility>
+
+#include "BenchmarkMetadata.h"
+#include "util/json.h"
+#include "../benchmark/Benchmark.h"
+
+/*
+ * @brief Create a nlohmann::json object with all relevant informations
+ *  about the measurments taken by a BenchmarkRecords.
+ */
+nlohmann::json benchmarkRecordsToJson(const BenchmarkRecords& records);
```

Comment:
Maybe just write a custom converter for `BenchmarkRecords`. This spreads the functionality of the class a bit far apart.

Reply:
Yes, good idea, make this a to_json (no from_json needed), and declare it directly in the `.h` file of the `BenchmarkRecords`.
You can still leave the implementation in a dedicated `.cpp` file where it currently is, as they are all linked together.

Reply:
Done, see [8710005](https://github.com/ad-freiburg/qlever/pull/860/commits/87100050d06c033bd72ad5931b74fc2c338e24e5).